### DISCUSSION TITLE
How to find many by relation (e.g. find genres of artist by ID)

### DISCUSSION THREAD

**Discussion by vladinator1000** [prisma/prisma#10814] [Q&A] [answered]
Hi there, imagine you have `Artist`s and `Genre`s. How would you get the genres of an artist?

```prisma
model Artist {
  id     Int      @id @default(autoincrement())
  name   String
  genres Genre[]  @relation("artist_genre")
}

model Genre {
  id      Int      @id @default(autoincrement())
  name    String
  artists Artist[] @relation("artist_genre")
}
```

Let's say we have 10 genres. When I create an artist with 3 genres and get its genres like this
```ts
const genres = prisma.genre.findMany({
     where: {
        artists: {
          every: {
            id: artistId
           }
         }
      }
})
```

I expect to get 3 genres back, but I get all of them (10).

I found this workaround, by looking at the join table:

<img width="336" alt="TablePlus_ooenDDGCEE" src="https://user-images.githubusercontent.com/1226564/147001939-4c239294-ea73-4680-9853-207dc065fe1c.png">

```ts
const genreIds: { B: number }[] = await this.prisma
      .$queryRaw`SELECT "B" FROM "_artist_genre" WHERE "A" = ${artistId}`

const genres = await prisma.genre.findMany({
  where: {
    id: {
      in: genreIds.map((genre) => genre.B),
    },
  },
})
```

I would like to learn how to do this the proper way. What am I doing wrong here? I'm a bit confused by the `some`, `every`, `none` filters

**Comment by deckchairlabs** [comment, marked as answer]
```js
const genres = await prisma.artist.findUnique({
    where: {
        id: artistId
    }
}).genres({
    orderBy: {
        name: 'asc'
    }
})
```

**Reply by vladinator1000** [reply]
Oooh, this is cool! I didn't know you can chain them like that.

**Reply by deckchairlabs** [reply]
Yeah! I believe it's only the direct relations of the initial query. But yes, very handy!

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
